iceoryx_posh/test/moduletests/test_roudi_memory_manager.cpp

```diff
@@ -71,8 +71,21 @@ TEST_F(RouDiMemoryManager_Test, CallingCreateMemoryWithMemoryProviderSucceeds)
 
     EXPECT_THAT(sut.createAndAnnounceMemory().has_error(), Eq(false));
 
-    EXPECT_CALL(memoryBlock1, destroyMock());
-    EXPECT_CALL(memoryBlock2, destroyMock());
+//    EXPECT_CALL(memoryBlock1, destroyMock());
+//    EXPECT_CALL(memoryBlock2, destroyMock());
+}
+
+TEST_F(RouDiMemoryManager_Test, CallingCreateMemoryWithMemoryProviderError)
+{
+    sut.addMemoryProvider(&memoryProvider1);
```

Comment:
Why does this lead to an error? I am not fully in the loop functionality-wise, a short explanation would be nice.

Reply:
have a look at the line below ;)

Reply:
I did, before the comment (I think you mean the ASSERT). It is still not clear to me why adding a ``memoryProvider1`` must lead to an error and neither the test name nor the error code are really providing this information. Maybe I have to take a look at ``memoryProvider1``.

Reply:
It doesn't make sense to have a memory provider without a memory block because the provider calculates the required memory from the contained blocks.
If no memory block were added to the memory provider it's usually an error, because the developer forgot to add one.